### PR TITLE
feat(maps): updated to ESRI arcgis-core v4.20.2

### PR DESCRIPTION
**Pull request by mstankala**
[12.4.0]:
* **maps:** updated to ESRI [@arcgis/core v4.20.2](https://www.npmjs.com/package/@arcgis/core/v/4.20.2)
* **maps:** example WebScene `portalItem` changed 
* **maps:** `SbbEsriConfiguration.originsWithCredentialsRequired` property marked as deprecated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
+## [12.4.0](https://github.com/sbb-design-systems/sbb-angular/compare/12.3.0...12.4.0) (2021-09-02)
```
Comment (kyubisation) [inline CHANGELOG.md:5]: Changelog will be automatically generated based on merged PRs (or other commits on master).
Please do not change it.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sbb-angular",
-  "version": "12.3.0",
+  "version": "12.4.0",
```
Comment (kyubisation) [inline package.json:3]: Do not change the version.

File: package.json [hunk 3]
```diff
@@ -66,7 +66,7 @@
     "@angular/platform-browser": "~12.2.0",
     "@angular/platform-browser-dynamic": "~12.2.0",
     "@angular/router": "~12.2.0",
-    "@arcgis/core": "~4.19.1",
+    "@arcgis/core": "^4.20.2",
```
Comment (kyubisation) [inline package.json:69]: Should this not use `~` instead of `^`?

File: src/angular-maps/getting-started.md [hunk 4]
```diff
@@ -76,17 +77,32 @@ export class TrainChooChooAppModule { }
...
+  imports: [SbbEsriConfigModule],
   ...
 })
 export class TrainChooChooAppModule {}
 ```
 
-### Step 3: Import the component modules
+### Step 3: Initialize the library configuration
+
+In your app.component.ts file (or any startup service), import the `SbbEsriLoaderService` and add as constructor parameter. Otherwise the configuration (the standard or your specific) won't initialize:
```
Comment (kyubisation) [inline src/angular-maps/getting-started.md:92]: Why is this necessary?
This seems to be better suited for a `forRoot` configuration.

**Review by kyubisation — CHANGES_REQUESTED** [review]
Thank you for the PR. 👍 

Please do not change the version or the changelog.
Be aware that this PR will be squashed into one commit.

File: src/angular-maps/getting-started.md (hunk 4 above)
Comment (mstankala) [reply src/angular-maps/getting-started.md:92]: SbbEsriLoaderService is using this config (injected with SBB_ESRI_CONFIG_TOKEN). But SbbEsriLoaderService is not "injected" anywhere, so it's not "instantiated" automatically. Do you have any Idea how I can archive that?

Comment (kyubisation) [reply src/angular-maps/getting-started.md:92]: You should be able to instantiate the service directly in the forRoot method.

```ts

@NgModule({
  imports: [CommonModule],
  declarations: []
})
export class ExampleModule {
  static forRoot(toggle: boolean): ModuleWithProviders<ExampleModule> {
    return {
      ngModule: ExampleModule,
      providers: [
        {
          provide: SimpleService,
          useValue: new SimpleService({ message: 'example', toggle })
        }
      ]
    };
  }
}

```

https://stackblitz.com/edit/angular-ivy-cf3o73?file=src%2Fapp%2Fexample%2Fexample.module.ts

Alternatively you should also be able to apply this configuration in the forRoot method, although I haven't tested this yet.

File: src/angular-maps/esri-config/esri-config.module.ts [hunk 5]
```diff
@@ -1,17 +1,19 @@
 import { ModuleWithProviders, NgModule } from '@angular/core';
 
-import { SBB_ESRI_CONFIG_TOKEN } from './esri-config.token';
+import { SbbEsriConfigConsts } from '../esri-config/esri-standard-values.const';
```
Comment (kyubisation) [inline src/angular-maps/esri-config/esri-config.module.ts:3]:
```suggestion
import { SbbEsriConfigConsts } from './esri-standard-values.const';
```

Comment (mstankala) [reply src/angular-maps/esri-config/esri-config.module.ts]: Done.

**Review by kyubisation — APPROVED** [review]
Thank you for the fixes.
LGTM 👍

**Comment by mstankala** [comment]
@kyubisation  Can you help me with that build error?
`
ERROR: /home/runner/work/sbb-angular/sbb-angular/src/angular-maps/esri-config/BUILD.bazel:11:10: Compiling Angular templates (ViewEngine - prodmode) //src/angular-maps/esri-config:esri-config failed: (Exit 1): ngc-wrapped.sh failed: error executing command bazel-out/host/bin/external/npm/@angular/bazel/bin/ngc-wrapped.sh '--node_options=--expose-gc' '--node_options=--stack-trace-limit=100' '--node_options=--max-old-space-size=4096' ... (remaining 1 argument(s) skipped)
`

**Comment by kyubisation** [comment]
@mstankala I'll have a look. 
Locally you can test this by running `bazel build src/angular-maps/esri-config --config=view-engine`.

File: src/angular-maps/esri-config/esri-loader.service.spec.ts [hunk 6]
```diff
@@ -1,42 +1,27 @@
...
     portalUrl: 'urlToPortalInstance',
-    originsWithCredentialsRequired: ['o1', 'o2'],
   };
 
-  let loaderService: SbbEsriLoaderService;
   const configureTestingModule = (customConfig?: SbbEsriConfiguration) => {
     TestBed.configureTestingModule({
-      imports: [...(customConfig ? [SbbEsriConfigModule.forRoot(customConfig!)] : [])],
-      providers: [SbbEsriLoaderService],
+      imports: [...[SbbEsriConfigModule.forRoot(customConfig)]],
```
Comment (kyubisation) [inline src/angular-maps/esri-config/esri-loader.service.spec.ts:14]:
```suggestion
      imports: [SbbEsriConfigModule.forRoot(customConfig)],
```

**Comment by kyubisation** [comment]
At this point `SbbEsriLoaderService` no longer serves any purpose as a service.
Would it maybe be better to implement the configuration as a conditional app initializer function: https://angular.io/api/core/APP_INITIALIZER

**Comment by mstankala** [comment]
@kyubisation All done 👍
